sauna/__init__.py

```diff
@@ -134,76 +156,72 @@ def launch_all_checks(plugins_config, hostname):
             )
             yield s
 
+    def run_producer(self):
+        while True:
+            for service_check in self.launch_all_checks(self.hostname):
+                logging.debug(
+                    'Pushing to main queue: {}'.format(service_check))
+                self.send_data_to_consumers(service_check)
+            if self.must_stop.wait(timeout=self.periodicity):
+                break
+        logging.debug('Exited producer thread')
+
+    def run_consumer(self, consumer_name, consumer_config, consumer_queue):
+        logging.debug(
+            'Running {} with {}'.format(consumer_name, consumer_config))
+        try:
+            consumer = consumers.get_consumer(consumer_name)(consumer_config)
+        except DependencyError as e:
+            print(str(e))
+            exit(1)
 
-q = queue.Queue()
-must_stop = threading.Event()
-try:
-    # In Python 3.2 threading.Event is a factory function
-    # the real class lives in threading._Event
-    event_type = threading._Event
-except AttributeError:
-    event_type = threading.Event
+        while not self.must_stop.is_set():
+            service_check = consumer_queue.get()
+            logging.debug(
+                'consumer {}  get {}'.format(consumer_name, service_check))
```

Comment:
And can you place it after the isinstance, to avoid:
```
2016/03/17 14:45:09 - DEBUG    - consumer Stdout  get <threading._Event object at 0x17f1c90>
```